consensus/src/payload_manager.rs

```diff
@@ -121,82 +133,114 @@ impl PayloadManager {
 
     /// Extract transaction from a given block
     /// Assumes it is never called for the same block concurrently. Otherwise status can be None.
-    pub async fn get_transactions(&self, block: &Block) -> ExecutorResult<Vec<SignedTransaction>> {
+    pub async fn get_transactions(
+        &self,
+        block: &Block,
+    ) -> ExecutorResult<(Vec<SignedTransaction>, usize)> {
         let payload = match block.payload() {
             Some(p) => p,
-            None => return Ok(Vec::new()),
+            None => return Ok((Vec::new(), usize::MAX)),
         };
 
+        async fn process_payload(
+            proof_with_data: &ProofWithData,
+            batch_reader: Arc<dyn BatchReader>,
+            block: &Block,
+            max_txns_to_include_in_block: usize,
+        ) -> ExecutorResult<(Vec<SignedTransaction>, usize)> {
```

Comment:
I think it will be cleaner to not pass max_txns_to_include_in_block to this function at all (as we are only returning it), but to be creating a tuple outside, and have this return just `ExecutorResult<Vec<SignedTransaction>>`

Reply:
Done